### PR TITLE
Fix server crash on server-window resize

### PR DESCRIPTION
**Pull request by Necktrox**
This pull request patches the server implementation to use a vector instead of a fixed-width array, because the method `ShowInfoTag` used to write beyond the array boundaries if you resize the server window corrupting the memory of the command queue and further.

I may not have enough experience when it's about C++, performance and vectors. Please tell me your thoughts about the code below (as review or comment).

### CODE REVIEW

**Comment by qaisjp** [comment]
This is still pending review, and clashes with this: https://github.com/multitheftauto/mtasa-blue/commit/f565c40e9bba67067749f7574f8c3fcc057ca76f

Once this has been reviewed, please can you rebase?

**Comment by Necktrox** [comment]
Rebased on top of master.

**Comment by qaisjp** [comment]
Note to reader: [setting whitespace to 1 helps make this PR more readable](https://github.com/multitheftauto/mtasa-blue/pull/86/files?w=1).

File: Server/core/CThreadCommandQueue.h [hunk 1]
```diff
@@ -25,10 +25,8 @@ class CThreadCommandQueue
     void                Process         ( bool& bRequestedQuit, class CModManagerImpl* pModManager );
 
 private:
-    std::list < std::string >       m_Commands;
-    CCriticalSection                m_Critical;
+    std::queue < std::string >  m_Commands;
```
Comment (qaisjp) [inline Server/core/CThreadCommandQueue.h]: Is the use of `std::queue` actually a contributing factor in fixing the crash?  (adding this is a good idea though!)

**Comment by 4O4** [comment]
@Necktrox it's not going to be merged?

**Comment by Necktrox** [comment]
A little commit with a working fix is underway

**Pull request closed without merging**